content/en/sensitive_data_scanner/regular_expression_syntax.md

```diff
@@ -0,0 +1,231 @@
+| ---- | ---------------- | ------------------------------------------------------------------------------------- |
+| `i`  | Case insensitive | Letters match both upper and lower case.                                              |
+| `m`  | Multi-line mode  | `^` and `$` match the beginning and end of line.                                      |
+| `s`  | Single line      | Allows `.` to match any character, when it usually matches anything except `\n`).     |
+| `x`  | Extended         | Whitespace is ignored (except in a custom character class).                           |
+
+## Quoting
+
+Use the regex syntax `\Q...\E` to treat everything between `\Q` and `\E` as a literal.
+
+
+## Quantifiers
+
+Quantifiers repeat the previous expression. Greedy means that the most number of repetitions are taken, and are only given back as needed to find a match. Lazy takes the minimum number of repetitions and adds more as needed.
+
+| Regex Syntax | Description                                                                   |
+| ------------ | ----------------------------------------------------------------------------- |
+| `?`          | Repeat `0` or `1` time (greedy).                                              |
+| `??`         | Repeat `0` or `1` time (lazy).                                                |
+| `+`          | Repeat `1` or more times (greedy).                                            |
+| `+?`         | Repeat `1` or more times (lazy).                                              |
+| `*`          | Repeat `0` or more times (greedy).                                            |
+| `*?`         | Repeat `0` or more times (lazy).                                              |
+| `{n}`        | Repeat exactly `n` times (the lazy modifier is accepted here but is ignored). |
+| `{n,m}`      | Repeat at least `n` times, but no more than `m` times (greedy).               |
+| `{n,m}?`     | Repeat at least `n times`, but no more than `m` times (lazy).                 |
+| `{n,}`       | Repeat at least `n` times (greedy).                                           |
+| `{n,}?`      | Repeat at least `n` times (lazy).                                             |
+
+**Note**: `{,m}`is not valid and is treated as a literal. Similarly, any syntax differences such as adding spaces inside the braces cause the quantifier to be treated as a literal instead.
+
+## Unicode properties
+
+Unicode properties for `x` in the character class `\p{x}`.
+
+| Unicode Properties| Description           |
+| ------------------| --------------------- |
+| `C`               | Other                 |
+| `Cc`              | Control               |
+| `Cf`              | Format                |
+| `Cn`              | Unassigned            |
+| `Co`              | Private use           |
+| `Cs`              | Surrogate             |
+|                   |                       |
```

Comment:
The empty rows seemed a bit odd to me at first, I thought some values were missing. I'd personally prefer to just not have the extra spacing unless this is common in other places (or maybe there's another way to add a separator?). Just a preference here, feel free to ignore it.

Reply:
Ah makes sense, I removed them. We don't do that anywhere else in the docs afaik.